### ISSUE TITLE
smart labeler feature

### ISSUE THREAD

**Issue by joshknnd1982** [nvaccess/nvda#7079]
In the audex app for android, it has a smart labeler feature where it digs into the underlying information behind an unlabeled or unknown object or button and autolabels that object or button. example: without audex running to clear all apps I just hear unlabeled button. with audex running and smart autolabeler turned on I then hear, clear recent apps button. so could you implement a similar smart label feature into NVDA where NVDA digs into unknown and unlabeled objects and tries to label them and to know what they are and how to use them. This would be a quite useful feature to make many more applications useable.

**Comment by Brian1Gaff** [comment]
If that info was actually available then one would imagine it might be an 
add on already. I fancy for many of these application, the information is 
not attainable except by trial and error of the user, and I thought there 
was a labelling add on somewhere, or did I dream that.

The first cause of action is always to attempt to tell the authors of 
course.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-

**Comment by jage9** [comment]
I agree an add-on may be the best place for this, at least in the experimentation phase. It could be a valuable addition down the road however.

**Comment by feerrenrut** [comment]
At the moment NVDA does its best to label controls. Within NVDA there is a concept of an `AppModule`, this allows someone (typically a developer) to work out what a control should be, and how to identify it, and provide that information.

However to create these `AppModules` NV Access relies on specific examples of application controls that need further investigation, or on developers to contribute the code for the AppModule to NVDA.

In short, the mechanism is there, if there are specific examples, please raise a new issue for each.

Another similar request is for NVDA users to label controls / objects themselves, for this see #2111 

Since there does not seem to be any specific action we can take on this issue, I am going to close it.
